module/VuFindConsole/tests/unit-tests/src/VuFindTest/Command/Util/UpdateResourceMetadataCommandTest.php

```diff
@@ -77,22 +82,59 @@ public static function updateProvider(): array
     #[\PHPUnit\Framework\Attributes\DataProvider('updateProvider')]
     public function testUpdate(array $commandParams, array $expectedFindParams)
     {
+        $resource = $this->createMock(ResourceEntityInterface::class);
+        $resource->expects($this->once())
+            ->method('getId')
+            ->willReturn(123);
+        $resource->expects($this->exactly(2))
+            ->method('getRecordId')
+            ->willReturn('foo');
+        $resource->expects($this->exactly(2))
+            ->method('getSource')
+            ->willReturn('src');
         $resourceService = $this->createMock(ResourceServiceInterface::class);
-        $resourceService->expects($this->once())
-            ->method('findMetadataToUpdate')
-            ->with(...$expectedFindParams)
-            ->willReturn([]);
+
+        $secondFindParams = $expectedFindParams;
+        $secondFindParams[0] = 123;
+        $this->expectConsecutiveCalls(
+            $resourceService,
+            'findMetadataToUpdate',
+            [
+                $expectedFindParams,
+                $secondFindParams,
+            ],
+            [
+                [$resource],
+                [],
+            ]
+        );
+
+        $driver = $this->createMock(DefaultRecord::class);
+        $driver->expects($this->any())
+            ->method('getUniqueID')
```

Comment:
We should drop the "any" expectation; it's not really meaningful, and it would be automatically removed when this was merged to dev due to recent Rector changes in #4948.

```suggestion
        $driver->method('getUniqueID')
```

Reply:
Alternatively, if you want to have a specific expectation here, why not assign `$driverRecordId` before the assert call instead of in the else clause, and then you can assert on the variable and know for certain how many times `getUniqueID` will be called?

Reply:
Good idea, done!